### PR TITLE
chore(broker): Remove the legacy TOML configuration module

### LINKED ISSUE #4121: Remove Support for Legacy Environment Variables in 0.24.0
**Description**

With release 0.23.0 the TOML based configuration was replaced with YAML based configuration. This also resulted in a change to environment variables to configure the system.

The old environment variables are still supported for backwards portability in 0.23.0. This support is scheduled to be removed in 0.24.0.

This entails the following steps:
* Remove maven module `zeebe-legacy-toml-config`
* Clean up callers of this module
* Remove corresponding sections in `deprecated-features.md`

### PR DESCRIPTION
**Pull request by aivinog1**
## Description

Remove the legacy TOML configuration module

## Related issues

closes #4121

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [x] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [x] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by pihme** [comment]
Looks good on the first glance. Created a branch to run all tests on it. Will give it a more thorough look later today.

**Comment by pihme** [comment]
All tests passed. The failure mentioned above is just a flaky test.

**Comment by aivinog1** [comment]
@pihme Looks great, thanks :)

File: docs/src/appendix/deprecated-features.md [hunk 1]
```diff
@@ -6,20 +6,6 @@ This section lists deprecated features.
 - TOML configuration - deprecated and removed in 0.23.0-alpha2
 - Legacy environment variables - deprecated in 0.23.0-alpha2, to be removed in 0.24.0
```
Comment (pihme) [inline docs/src/appendix/deprecated-features.md:7]:
```suggestion
- Legacy environment variables - deprecated in 0.23.0-alpha2, removed in 0.25.0-alpha2
```

File: docs/src/appendix/deprecated-features.md [hunk 2]
```diff
@@ -31,7 +17,7 @@ exporters:
 
 In terms of specifying values, there were two minor changes:
 - Memory sizes are now specified like this: `512MB` (old way: `512M`)
-- Durations, e.g. timeouts, can now also be given in ISO-8601 Durations format. However you can still use the established way and specify a timeout of `30s`  
+- Durations, e.g. timeouts, can now also be given in ISO-8601 Durations format. However you can still use the established way and specify a timeout of `30s`
 
 ### Legacy Environment Variables
```
Comment (pihme) [inline docs/src/appendix/deprecated-features.md:22]: This section needs to be removed. The legacy environment variables were mapped in the TOML configuration module.

**Review by pihme — CHANGES_REQUESTED** [review]
Good work. 

Some details could be improved.

**Comment by pihme** [comment]
@npepinpe just to let you know that this PR will remove legacy support for old environment variables (which maybe isn't too obvious from the title). 
If I don't get a veto from you, I will assume this is ok.

File: docs/src/appendix/deprecated-features.md (hunk 1 above)
Comment (aivinog1) [reply docs/src/appendix/deprecated-features.md]: Good point, done :)

File: docs/src/appendix/deprecated-features.md (hunk 2 above)
Comment (aivinog1) [reply docs/src/appendix/deprecated-features.md]: Done :)

**Comment by aivinog1** [comment]
@pihme Hi, I've fixed all review comments :)

**Review by pihme — APPROVED** [review]
Good work!

Thanks for your contribution

**Comment by aivinog1** [comment]
@pihme I found out that release of 0.25.0-alpha2 already issued (https://github.com/zeebe-io/zeebe/releases/tag/0.25.0-alpha2), but commit for this PR isn't included in that release. So, now there is a documentation issue :(

**Pull request merged**